sstables/sstable_set.hh

```diff
@@ -30,8 +32,15 @@ class incremental_selector_impl;
 // SStables in that same run will not overlap with one another.
 class sstable_run {
     sstable_list _all;
+
+    using interval_set_type = boost::icl::interval_set<compatible_ring_position_or_view>;
+    using interval_type = interval_set_type::interval_type;
+    interval_set_type _interval_set;
+private:
```

Comment:
We avoided so far polluting headers with interval_set, please keep it out by using indirection.

interval_set is overkill here (since we'll never have overlaps actually in the set), but okay.

Reply:
A price to pay for keeping the invariant, right?  Also provides us with O(log N) on overlapping detection. I'll have to extend this once I split large partitions, so there can be overlapping for a given key, but not at clustering dimension.

Will fix the pollution in v2. thanks.

Reply:
It's possible to check the invariant with O(log n) work, without interval_set, but it requires a bit more work.

Reply:
Challenge accepted.